pkgs/applications/editors/oni2/default.nix

```diff
@@ -11,6 +11,6 @@ in mkOni2 rec {
   version = "0.5.7";
   rev = "v${version}";
   sha256 = "NlN0Ntdwtx5XLjd1ltUzv/bjmJQR5eyRqtmicppP6YU=";
-  fetchDepsSha256 = "k7G6jPJfxCCSuSucPfiXljCVJhmjl/BxWMCEjv2tfhA=";
+  fetchDepsSha256 = "jtPRgAFyoy2ILPQyHVVIzX2DZj30K6xnJei1P4SMtBc=";
```

Comment:
It seems to fail for me:

```
error: hash mismatch in fixed-output derivation '/nix/store/zwfppn42k5i6f2cvbv23ydlfg0wz86z0-oni2-fetch-deps.drv':
         specified: sha256-jtPRgAFyoy2ILPQyHVVIzX2DZj30K6xnJei1P4SMtBc=
            got:    sha256-k7G6jPJfxCCSuSucPfiXljCVJhmjl/BxWMCEjv2tfhA=
```

Reply:
not quite sure what to make of this... ftr the new hash fails for me too in this branch but passes in nixpkgs-review.

<details>
<summary>Output:</summary>
<pre>
❯ diffoscope /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps
--- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps
+++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps
│   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy
├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy
│ ├── file list
│ │ @@ -1,3 +1,3 @@
│ │  3
│ │ +3__________________________
│ │ -3_________________________________________________________________________
│ │  source
│ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source
│ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source
│ │ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source/i
│ │ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source/i
│ │ │ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source/i/lodash__4.17.21__82c45c9d
│ │ │ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source/i/lodash__4.17.21__82c45c9d
│ │ │ │ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source/i/lodash__4.17.21__82c45c9d/fp
│ │ │ │ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source/i/lodash__4.17.21__82c45c9d/fp
│ │ │ │ │ ├── stat {}
│ │ │ │ │ │ @@ -1,7 +1,7 @@
│ │ │ │ │ │  
│ │ │ │ │ │ +  Size: 16384      Blocks: 32         IO Block: 4096   directory
│ │ │ │ │ │ -  Size: 20480      Blocks: 40         IO Block: 4096   directory
│ │ │ │ │ │  Device: 8,32       Access: (0555/dr-xr-xr-x)  Uid: ( 1000/   smona)   Gid: ( 1000/   smona)
│ │ │ │ │ │  
│ │ │ │ │ │  Modify: 1970-01-01 00:00:01.000000000 +0000
│ │ │ │ │ ├── stat {}
│ │ │ │ │ │ @@ -1,7 +1,7 @@
│ │ │ │ │ │  
│ │ │ │ │ │ -  Size: 20480      Blocks: 40         IO Block: 4096   directory
│ │ │ │ │ │ +  Size: 16384      Blocks: 32         IO Block: 4096   directory
│ │ │ │ │ │  Device: 8,32       Access: (0555/dr-xr-xr-x)  Uid: ( 1000/   smona)   Gid: ( 1000/   smona)
│ │ │ │ │ │  
│ │ │ │ │ │  Modify: 1970-01-01 00:00:01.000000000 +0000
│ │ │ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source/i/opam__s__omd__337d3e7d
│ │ │ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source/i/opam__s__omd__337d3e7d
│ │ │ │ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source/i/opam__s__omd__337d3e7d/tests
│ │ │ │ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source/i/opam__s__omd__337d3e7d/tests
│ │ │ │ │ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source/i/opam__s__omd__337d3e7d/tests/commonmark
│ │ │ │ │ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source/i/opam__s__omd__337d3e7d/tests/commonmark
│ │ │ │ │ │ ├── stat {}
│ │ │ │ │ │ │ @@ -1,7 +1,7 @@
│ │ │ │ │ │ │  
│ │ │ │ │ │ │ -  Size: 45056            Blocks: 88         IO Block: 4096   directory
│ │ │ │ │ │ │ +  Size: 36864            Blocks: 72         IO Block: 4096   directory
│ │ │ │ │ │ │  Device: 8,32     Access: (0555/dr-xr-xr-x)  Uid: ( 1000/   smona)   Gid: ( 1000/   smona)
│ │ │ │ │ │ │  
│ │ │ │ │ │ │  Modify: 1970-01-01 00:00:01.000000000 +0000
│ │ │ │ │ │ ├── stat {}
│ │ │ │ │ │ │ @@ -1,7 +1,7 @@
│ │ │ │ │ │ │  
│ │ │ │ │ │ │ +  Size: 36864            Blocks: 72         IO Block: 4096   directory
│ │ │ │ │ │ │ -  Size: 45056            Blocks: 88         IO Block: 4096   directory
│ │ │ │ │ │ │  Device: 8,32     Access: (0555/dr-xr-xr-x)  Uid: ( 1000/   smona)   Gid: ( 1000/   smona)
│ │ │ │ │ │ │  
│ │ │ │ │ │ │  Modify: 1970-01-01 00:00:01.000000000 +0000
│ │ │ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5
│ │ │ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5
│ │ │ │ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5/harfbuzz-2.6.
8
│ │ │ │ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5/harfbuzz-2.6.
8
│ │ │ │ │ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5/harfbuzz-2.
6.8/test
│ │ │ │ │ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5/harfbuzz-2.
6.8/test
│ │ │ │ │ │ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5/harfbuzz-
2.6.8/test/subset
│ │ │ │ │ │ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5/harfbuzz-
2.6.8/test/subset
│ │ │ │ │ │ │ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5/harfbuz
z-2.6.8/test/subset/data
│ │ │ │ │ │ │ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5/harfbuz
z-2.6.8/test/subset/data
│ │ │ │ │ │ │ │ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5/harfb
uzz-2.6.8/test/subset/data/expected
│ │ │ │ │ │ │ │ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5/harfb
uzz-2.6.8/test/subset/data/expected
│ │ │ │ │ │ │ │ │ │   --- /nix/store/fkyri627k46clh3mqy9vq9qxbnprzsdx-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5/har
fbuzz-2.6.8/test/subset/data/expected/cbdt
│ │ │ │ │ │ │ │ │ ├── +++ /nix/store/dvzk5xy24jvqpkmaibw9zfp89yh7xqlc-oni2-fetch-deps/esy/source/i/revery__s__esy_harfbuzz__2.6.8002__3348a4f5/har
fbuzz-2.6.8/test/subset/data/expected/cbdt
│ │ │ │ │ │ │ │ │ │ ├── stat {}
│ │ │ │ │ │ │ │ │ │ │ @@ -1,7 +1,7 @@
│ │ │ │ │ │ │ │ │ │ │  
│ │ │ │ │ │ │ │ │ │ │ -  Size: 12288            Blocks: 24         IO Block: 4096   directory
│ │ │ │ │ │ │ │ │ │ │ +  Size: 16384            Blocks: 32         IO Block: 4096   directory
│ │ │ │ │ │ │ │ │ │ │  Device: 8,32     Access: (0555/dr-xr-xr-x)  Uid: ( 1000/   smona)   Gid: ( 1000/   smona)
│ │ │ │ │ │ │ │ │ │ │  
│ │ │ │ │ │ │ │ │ │ │  Modify: 1970-01-01 00:00:01.000000000 +0000
│ │ │ │ │ │ │ │ │ │ ├── stat {}
│ │ │ │ │ │ │ │ │ │ │ @@ -1,7 +1,7 @@
│ │ │ │ │ │ │ │ │ │ │  
│ │ │ │ │ │ │ │ │ │ │ +  Size: 16384            Blocks: 32         IO Block: 4096   directory
│ │ │ │ │ │ │ │ │ │ │ -  Size: 12288            Blocks: 24         IO Block: 4096   directory
│ │ │ │ │ │ │ │ │ │ │  Device: 8,32     Access: (0555/dr-xr-xr-x)  Uid: ( 1000/   smona)   Gid: ( 1000/   smona)
│ │ │ │ │ │ │ │ │ │ │  
│ │ │ │ │ │ │ │ │ │ │  Modify: 1970-01-01 00:00:01.000000000 +0000
</pre>
</details>

Reply:
Update: I created a copy of this branch and rebased it to master as of this comment. Then the build succeeds with the new hash. I guess nixpkgs-review works!

(i also accidentally synced this branch on my fork rather than master first, so force-pushed to undo that merge commit)

Reply:
It think that hash mismatch might be unrelated to this change and the issue might be that the package is non-deterministic.

Reply:
i think you're right. so, does the PR look good to go now?

Reply:
I would drop the oni2 change.

Reply:
done :heavy_check_mark: